spec/ddtrace/contrib/que/tracer_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'ddtrace/contrib/support/spec_helper'
+require 'ddtrace/contrib/analytics_examples'
+require 'ddtrace'
+require 'que'
+
+RSpec.describe Datadog::Contrib::Que::Tracer do
+  let(:que_tracer) { described_class.new }
+  let(:configuration_options) { {} }
+
+  class TestJobClass < ::Que::Job
+    def run(*args); end
+  end
```

Comment:
Declaring a class here actually declares it on the global namespace, which is shared with all test cases.

On alternative we normally take in these cases is to either:
1. Not declare a class constant if not needed for the test:
```ruby
let(:job_class) do
  Class.new(::Que::Job) do
    def run(*args); end
  end
end
```
2. If a class constant is needed:
```ruby
before do
  stub_const('TestJobClass', Class.new(::Que::Job) do
    def run(*args); end
  end)
end
```

One reason a class constant might be needed is when a code path tries to read the class name: `job_class.name` will return `nil`, while `TestJobClass.name` will return `"TestJobClass"`.

Reply:
👍 makes sense.

I am going to have to carve out some this weekend to see if I can make a headway in the tests here. Que is giving me some hard time. Need to understand the inner workings of the gem a bit better to manipulate jobs in test setup. (the right mock setup.)

Objective: 
- Find a way to populate jobs so that I can assert something about each span I am collecting data for
- cover failure cases for when things go wrong and etc..

Reply:
I have been doing something wrong initially but when I figured out my mistake `Que::Job.run_synchronously = true` worked like a charm.